static/js/l10n/i18n.ts

```diff
@@ -0,0 +1,55 @@
+import { createIntl, createIntlCache } from "react-intl";
+
+// Get locale from query params. If none, use navigator language.
+function determineLocale() {
+  const urlParams = new URLSearchParams(window.location.search);
+  let locale = urlParams.get("hl");
+  if (!locale) {
+    locale = navigator.language;
+  }
+  return locale;
+}
+const locale = determineLocale();
+
+// A single cache instance can be shared for all locales.
+const intlCache = createIntlCache();
+
+// TODO(tjann): see if we can ship locale specific js bundles.
+function loadLocaleData(locale: string): Promise<Record<any, any>> {
+  switch (locale) {
+    case "es":
+      return import("./compiled-lang/es.json");
+    default:
+      return import("./compiled-lang/en.json");
+  }
+}
+
+// This IntlShape object will be used for both React Intl's
+// React Component API (arg for RawIntlProvider) and
+// Imperative API (format<X> method).
+let intl;
```

Comment:
this might have to be a promise, or this could end up used uninitialized. add a todo?